frontend/hooks/useNeedsFunds.ts

```diff
@@ -28,35 +24,38 @@ export const useNeedsFunds = (serviceConfigId: string) => {
     [service],
   );
 
-  const serviceFundRequirements = useMemo< {
+  const serviceFundRequirements = useMemo<{
     [chainId: number]: {
-      [tokenSymbol: string]: number;    
-    }
+      [tokenSymbol: string]: number;
+    };
   }>(() => {
     if (!serviceTemplate) return {};
 
     const results: {
       [chainId: number]: {
-        [tokenSymbol: string]: number;    
-      }
+        [tokenSymbol: string]: number;
+      };
     } = {};
 
     Object.entries(serviceTemplate.configurations).forEach(
       ([chainId, config]) => {
-        const serviceTemplateDefault = serviceTemplate.configurations[+chainId].staking_program_id
-        const serviceCurrent = service?.chain_configs[+chainId]?.chain_data?.user_params?.staking_program_id
-
-        if (!serviceCurrent && !serviceTemplateDefault) return;
-
+        const templateStakingProgramId =
+          serviceTemplate.configurations[+chainId].staking_program_id;
+        const serviceStakingProgramId =
+          service?.chain_configs[+chainId]?.chain_data?.user_params
+            ?.staking_program_id;
+        const stakingProgramId =
+          serviceStakingProgramId ?? templateStakingProgramId;
+
+        if (!stakingProgramId) return;
         if (!service?.chain_configs[+chainId]) return;
+
         const gasEstimate = config.monthly_gas_estimate;
         const monthlyGasEstimate = Number(formatUnits(`${gasEstimate}`, 18));
         const minimumStakedAmountRequired =
-          STAKING_PROGRAMS[+chainId][
-            service?.chain_configs[+chainId]?.chain_data?.user_params
-              ?.staking_program_id ??
-              serviceTemplate.configurations[+chainId].staking_program_id
-          ].stakingRequirements.OLAS;
+          STAKING_PROGRAMS[+chainId]?.[stakingProgramId]?.stakingRequirements?.[
+            TokenSymbol.OLAS
+          ] || 0;
```

Comment:
had to add || 0 here b/c we map on `serviceTemplate.configurations` which for optimus will contain three chains, but we'll have `stakingRequirements` in STAKING_PROGRAM only for one chain leading to undefined on others.

maybe makes sense to reconsider the `serviceFundRequirements` logic, we need native tokens on different chains, but olas for staking only on one. but at the moment this approach feels the most logical